sv_mini_atlas/library/urn.py

```diff
@@ -84,19 +103,25 @@ def to_no_passage(self):
     def up_to(self, key):
         if key == self.NO_PASSAGE:
             label = "no_passage"
+        elif key == self.NID:
+            label = "nid"
+        elif key == self.NAMESPACE:
+            label = "namespace"
         else:
             label = self.WORK_COMPONENT_LABELS.get(key, None)
         if label is None:
-            raise KeyError("Provided key is not recognized.")
+            raise KeyError(f"Provided key is not recognized: {key}")
 
         attr_name = f"to_{label}"
         try:
             value = getattr(self, attr_name)
         except TypeError:
-            raise ValueError(f'URN has no "{label}" component')
+            raise ValueError(f"URN has no component: {label}")
 
-        if key == self.NO_PASSAGE and self.parsed["work"]:
-            return value
+        # TODO: Is there any time we don't want a trailing ":" for any URN
```

Comment:
@jacobwegner this is my only outstanding 'not sure'.

Reply:
@jhrr:

I think we always want the trailing `:` on any work-component level URN.

Once we have a passage component leave URN, we never have the trailing colon.

Does that help to clarify?

Reply:
@jacobwegner yep that was also my thinking, so I think the [lines I commented](https://github.com/scaife-viewer/sv-mini-atlas/pull/3/files#diff-f059ee644c60f00cb5173d1aefb52c96R123-R124) can be removed altogether? I think we'll always be appending `:` if we use `up_to`?

Reply:
Agreed; this might have been cruft from the earlier backport I did.  Remove at will!